pallets/deitos-fs/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+        }
+
+        /// Unsigned call to submit the file validation from the offchain worker.
+        /// It checks the hash returned from the offchain worker and updates the file status if it matches.
+        /// In case not, it increases the error count.
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::submit_file_validation())]
+        pub fn submit_file_validation(
+            origin: OriginFor<T>,
+            file_id: T::FileId,
+            file: FileDetails<T>,
+            returned_hash: FileHash,
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            let mut new_file = file.clone();
+            if file.hash == returned_hash && file.status == FileValidationStatus::Pending {
+                new_file.status = FileValidationStatus::Verified;
+
+                Files::<T>::insert(file_id, new_file);
+                FilesToBeChecked::<T>::remove(file_id);
+                Self::deposit_event(Event::FileVerified { file_id });
+            } else {
+                FilesToBeChecked::<T>::mutate(file_id, |file_option| {
+                    if let Some(file_check) = file_option {
+                        let error_boundary: u32 = T::ErrorBoundary::get();
+                        file_check.error_count = file_check.error_count.saturating_add(1);
+
+                        if file_check.error_count > error_boundary {
+                            file_check.status = FileValidationStatus::Conflict;
+                            Self::deposit_event(Event::FileConflict { file_id });
+                        } else {
+                            Self::deposit_event(Event::FileNotVerified {
+                                file_id,
+                                error_count: file_check.error_count,
+                            });
+                        }
+                    }
+                });
+            }
+            Ok(())
+        }
+
+		#[pallet::call_index(2)]
+        #[pallet::weight(<T as Config>::WeightInfo::data_integrity_protocol())]
+        pub fn data_integrity_protocol(
+            origin: OriginFor<T>,
+            file_id: T::FileId, 
+            result: CheckResult
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            match result {
+                CheckResult::CheckPassed => {
+                    Self::deposit_event(Event::DataIntegrityCheckSuccessful { file_id });
+                },
+                CheckResult::DataIntegrityCheckFailed => {
+                    Self::deposit_event(Event::DataIntegrityCheckFailed { file_id });
+                }
+            }
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    fn fetch_file_hash(file_id: &str) -> Result<FileHash, http::Error> {
+        let deadline = sp_io::offchain::timestamp().add(Duration::from_millis(2_000));
+        // TODO: Replace this with the actual URL
+        let url = format!("http://localhost:3030/{}", file_id);
```

Comment:
The URL for the verifier in the docker file is going to be: http://verifier.deitos.network::4040